packages/sessions-sdk-react/src/session-state.ts

```diff
@@ -0,0 +1,120 @@
+import type { Session } from "@fogo/sessions-sdk";
+import type { MessageSignerWalletAdapterProps } from "@solana/wallet-adapter-base";
+import type { useConnection } from "@solana/wallet-adapter-react";
+import type { PublicKey } from "@solana/web3.js";
+
+export enum StateType {
+  Initializing,
+  CheckingStoredSession,
+  NotEstablished,
+  SelectingWallet,
+  WalletConnecting,
+  RequestingLimits,
+  SettingLimits,
+  Established,
+  UpdatingSession,
+  RequestingExtendedExpiry,
+  RequestingIncreasedLimits,
+}
+
+export type EstablishedOptions = Omit<Session, "sessionInfo"> & {
+  expiration: Date;
+  signMessage: (message: Uint8Array) => Promise<Uint8Array>;
+  createLogInToken: () => Promise<string>;
+  connection: ReturnType<typeof useConnection>["connection"];
+  isLimited: boolean;
+  endSession: () => void;
+  updateSession: (
+    prevState: StateType,
+    duration: number,
+    limits?: Map<PublicKey, bigint>,
+  ) => void;
+};
+
+export const SessionState = {
+  Initializing: () => ({ type: StateType.Initializing as const }),
```

Comment:
some of these feel extraneous? like do you need `Initializing`, or could that be recast to `UpdatingSession`? and there seems to be some leakage between session state and wallet state, or is that just a naming issue?

Reply:
> there seems to be some leakage between session state and wallet state

Regarding this, the coupling is very intentional -- the wallet state is in fact a portion of the session state machine.  We could technically break this up into a sub-machine and it might be marginally cleaner to do so, but since this is already and has been part of the public api for a long time I don't intend to change that public api with this PR